ouroboros/statequery/types.go

```diff
@@ -35,6 +34,22 @@ type UtxoTxID struct {
 	ID string `json:"id"`
 }
 
-type Ada struct {
-	Ada chainsync.Lovelace `json:"ada"`
+type Value struct {
+	Ada    int64
+	Assets map[string]map[string]int64
```

Comment:
You can make a type for the `Assets` type and call it something like `DoubleNestedInt64`. I use that already for `DoubleNestedInteger`.

Reply:
I don't like `DoubleNestedInteger` / `DoubleNestedInt64` for our type names.

Just define a single type, `type Value map[string]map[string]int64`; Ada is no longer a special case.

Reply:
It makes sense to privilege Ada in the record type because ada tends to be treated as a special case by other things, e.g. the corresponding value type in apollo has a separate field for ada. It would be slightly harder to construct the apollo types if the value was just one big map. How about `type Assets map[string]map[string]int64`?

Reply:
I don't think `Value` will work, at least without naming stuff in chainsync. (statequery may be totally different.) Over there, `Value` can have an extra field, and does in some cases.

Reply:
I'd prefer to cleave as close to possible to the ogmios types as possible;

The definition you have for Value is just a special case of `map[string]map[string]int64`, where `ada` and `lovelace` are keys; if you want a convenient way to read the ADA, you can define a method to extract the ADA like so:

```
const AdaPolicy = "ada"
const AdaName = "lovelace"
func (v Value) AdaLovelace() num.Int {
  return v.AssetAmount("ada.lovelace")
}
func (v Value) AssetAmount(asset AssetId) num.Int {
  if nested, ok := v[asset.PolicyId()]; ok {
    return nested[asset.AssetNameHex()]
  }
  return 0
}
```

Also, it's important not to use uint64; the reason we use num.Int is because the quantity of tokens can, in theory, exceed uint64 (not in a single mint, but if someone mints a token multiple times, then combines them in the same transaction output).

Reply:
The reason ADA is treated separately by Apollo is just because Blockfrost and Ogmios set a bad precedent of treating it separately.